File: app/src/main/java/mx/edu/upqroo/kristenandroid/data/database/daos/DayDao.java
package mx.edu.upqroo.kristenandroid.data.database.daos;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;
import mx.edu.upqroo.kristenandroid.data.database.entities.Day;

@Dao
public interface DayDao {

    @Insert
    long insert(Day day);

    @Update
    void update(Day... days);

    @Delete
    void delete(Day... days);

    @Query("DELETE FROM day")
    void deleteAll();

    @Query("SELECT * FROM day WHERE dayId = :dayId")
    LiveData<Day> getById(long dayId);

    @Query("SELECT * FROM day")
    LiveData<List<Day>> getAll();

    @Query("SELECT * FROM day WHERE userId = :userId")
    LiveData<List<Day>> getByUserId(final String userId);

    @Query("SELECT COUNT(*) FROM day")
    int count();
}

File: app/src/main/java/mx/edu/upqroo/kristenandroid/data/database/daos/DayDao.kt
package mx.edu.upqroo.kristenandroid.data.database.daos

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Transaction
import androidx.room.Update
import mx.edu.upqroo.kristenandroid.data.database.entities.Day
import mx.edu.upqroo.kristenandroid.data.models.ScheduleSubject

@Dao
interface DayDao {

    @get:Query("SELECT * FROM day")
    val all: LiveData<List<Day>>

    @Insert
    fun insert(day: Day): Long

    @Update
    fun update(vararg days: Day)

    @Delete
    fun delete(vararg days: Day)

    @Query("DELETE FROM day")
    fun deleteAll()

    @Query("SELECT * FROM day WHERE dayId = :dayId")
    fun getById(dayId: Long): LiveData<Day>

    @Query("SELECT * FROM day WHERE userId = :userId")
    fun getByUserId(userId: String): LiveData<List<Day>>

    @Query("SELECT COUNT(*) FROM day")
    fun count(): Int

    @Transaction
    @Query("SELECT * FROM day WHERE userId = :userId")
    fun getDaysAndSubjectsFromUser(userId: String): LiveData<List<ScheduleSubject>>

    @Transaction
    @Query("SELECT * FROM day WHERE userId = :userId")
    fun getDaysAndSubjectsFromUserSync(userId: String): List<ScheduleSubject>
}
